# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/other/DocAccessibleChild.cpp
# Commit: 6d0f4821ddcf
# Full Hash: 6d0f4821ddcfca1b59e4db3befc3ec9c12335891
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-03-23 09:49:32
# Regressor Bug: 1735970
# File Overlap Count: 1
# Description:
#   Bug 1735970 part 3: Add TableAccessibleBase and TableCellAccessibleBase. r=morgan
#   
#   These are needed to support RemoteAccessible tables.
#   Stuff specific to LocalAccessible is still in TableAccessible and TableCellAccessible, which now inherit from the new Base classes.
#   
# ==============================================================================

diff -r d82660fb5408 -r 6d0f4821ddcf accessible/ipc/other/DocAccessibleChild.cpp
--- a/accessible/ipc/other/DocAccessibleChild.cpp	Wed Mar 23 04:00:25 2022 +0000
+++ b/accessible/ipc/other/DocAccessibleChild.cpp	Wed Mar 23 04:00:26 2022 +0000
@@ -844,12 +844,11 @@
     const uint64_t& aID, nsTArray<uint64_t>* aCells) {
   TableCellAccessible* acc = IdToTableCellAccessible(aID);
   if (acc) {
-    AutoTArray<LocalAccessible*, 10> headerCells;
+    AutoTArray<Accessible*, 10> headerCells;
     acc->ColHeaderCells(&headerCells);
     aCells->SetCapacity(headerCells.Length());
-    for (uint32_t i = 0; i < headerCells.Length(); ++i) {
-      aCells->AppendElement(
-          reinterpret_cast<uint64_t>(headerCells[i]->UniqueID()));
+    for (Accessible* header : headerCells) {
+      aCells->AppendElement(header->ID());
     }
   }
 
@@ -860,12 +859,11 @@
     const uint64_t& aID, nsTArray<uint64_t>* aCells) {
   TableCellAccessible* acc = IdToTableCellAccessible(aID);
   if (acc) {
-    AutoTArray<LocalAccessible*, 10> headerCells;
+    AutoTArray<Accessible*, 10> headerCells;
     acc->RowHeaderCells(&headerCells);
     aCells->SetCapacity(headerCells.Length());
-    for (uint32_t i = 0; i < headerCells.Length(); ++i) {
-      aCells->AppendElement(
-          reinterpret_cast<uint64_t>(headerCells[i]->UniqueID()));
+    for (Accessible* header : headerCells) {
+      aCells->AppendElement(header->ID());
     }
   }
 
@@ -1106,11 +1104,11 @@
     const uint64_t& aID, nsTArray<uint64_t>* aCellIDs) {
   TableAccessible* acc = IdToTableAccessible(aID);
   if (acc) {
-    AutoTArray<LocalAccessible*, 30> cells;
+    AutoTArray<Accessible*, 30> cells;
     acc->SelectedCells(&cells);
     aCellIDs->SetCapacity(cells.Length());
-    for (uint32_t i = 0; i < cells.Length(); ++i) {
-      aCellIDs->AppendElement(reinterpret_cast<uint64_t>(cells[i]->UniqueID()));
+    for (Accessible* cell : cells) {
+      aCellIDs->AppendElement(cell->ID());
     }
   }
 